agents/kubevirt/fence_kubevirt.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/python -tt
+
+import os
+import sys, re
+import logging
+import atexit
+sys.path.append("/usr/share/fence/")
+from fencing import *
+from fencing import fail, fail_usage, run_delay, EC_STATUS, SyslogLibHandler
+from kubernetes import client, config
+from openshift.dynamic import DynamicClient
+from kubernetes.client.exceptions import ApiException
+
+API_VERSION='kubevirt.io/v1'
+
+logger = logging.getLogger("fence_kubevirt")
+logger.propagate = False
+logger.setLevel(logging.INFO)
+logger.addHandler(SyslogLibHandler())
+logging.getLogger('botocore.vendored').propagate = False
+
+def get_nodes_list(conn, options):
+    logger.info("Starting monitor operation")
+    result = {}
+    try:
+        namespace = options.get("--namespace")
+        include_uninitialized = True
+        vm_api = conn.resources.get(api_version=API_VERSION, kind='VirtualMachine')
+        vm_list = vm_api.get(namespace=namespace)
+        for vm in vm_list.items:
+            result[vm.metadata.name] = ("", None)
+    except Exception as e:
+        logger.error("Exception when calling VirtualMachine list: %s", e)
+    return result
+
+def get_power_status(conn, options):
+    logger.debug("Starting get status operation")
+    try:
+        namespace = options.get("--namespace")
+        name = options.get("--plug")
+        vmi_api = conn.resources.get(api_version=API_VERSION,
+                                              kind='VirtualMachineInstance')
+        vmi = vmi_api.get(name=name, namespace=namespace)
+        if vmi is not None:
+            phase = vmi.status.phase
+            if phase == "Running":
+                return "on"
+        return "off"
+    except ApiException as e:
+        if e.status == 404:
+            return "off"
+        logging.error("Failed to get power status, with API Exception: %s", e)
+        fail(EC_STATUS)
+    except Exception as e:
+        logging.error("Failed to get power status, with Exception: %s", e)
+        fail(EC_STATUS)
+
+def set_power_status(conn, options):
+    logger.debug("Starting set status operation")
+    try:
+        namespace = options.get("--namespace")
+        name = options.get("--plug")
+        action = 'start' if options["--action"] == "on" else 'stop'
+        virtctl_vm_action(conn, action, namespace, name)
+    except Exception as e:
+        logging.error("Failed to set power status, with Exception: %s", e)
+        fail(EC_STATUS)
+
+def define_new_opts():
+	all_opt["namespace"] = {
+		"getopt" : ":",
+		"longopt" : "namespace",
+		"help" : "--namespace=[namespace]        Namespace",
+		"shortdesc" : "Namespace.",
+		"required" : "1",
+		"order" : 2
+	}
+	all_opt["kubeconfig"] = {
```

Comment:
pardon my ignorance, but does this take envvar `KUBECONFIG` into account?

Reply:
You usually run the setup from one node and dont sync the profile to have it work, so using the envvar would mostly make sense for testing purposes.

Reply:
The kubernetes client use the three following options, by this order:

1. provided kubeconfig path (`--kubeconfig` flag)
2. KUBECONFIG Environment variable
3. Default path (`~/.kube/config`)

If `--kubeconfig` variable is not provided, it will try to use KUBECONFIG Environment variable